tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetReceiveVars.cpp

```diff
@@ -77,19 +77,23 @@ namespace {
 // correct arguments.
 template <typename InterpolationTargetTag>
 struct MockSendPointsToInterpolator {
-  template <
-      typename ParallelComponent, typename DbTags, typename Metavariables,
-      typename ArrayIndex,
-      Requires<tmpl::list_contains_v<
-          DbTags, intrp::Tags::TemporalIds<
-                      typename Metavariables::temporal_id::type>>> = nullptr>
+  template <typename ParallelComponent, typename DbTags, typename Metavariables,
+            typename ArrayIndex,
+            Requires<tmpl::list_contains_v<
+                DbTags, intrp::Tags::TemporalIds<
+                            typename Metavariables::InterpolationTargetA::
+                                temporal_id::type>>> = nullptr>
   static void apply(
       db::DataBox<DbTags>& box, Parallel::GlobalCache<Metavariables>& /*cache*/,
       const ArrayIndex& /*array_index*/,
-      const typename Metavariables::temporal_id::type& temporal_id) noexcept {
-    using temporal_id_type = typename Metavariables::temporal_id::type;
+      const typename Metavariables::InterpolationTargetA::temporal_id::type&
+          temporal_id) noexcept {
+    using temporal_id_type =
+        typename Metavariables::InterpolationTargetA::temporal_id::type;
     Slab slab(0.0, 1.0);
-    CHECK(temporal_id == TimeStepId(true, 0, Time(slab, Rational(14, 15))));
+    CHECK(temporal_id == TimeStepId(true, 0, Time(slab, Rational(14, 15)))
+                             .substep_time()
+                             .value());
```

Comment:
`14.0 / 15.0`

Same elsewhere, drop any unnecessary includes.  If you're concerned about roundoff you can switch to integral values or binary fractions.